### DISCUSSION TITLE
How to debug API calls?

### DISCUSSION THREAD

**Discussion by lostdog19** [serenity-js/serenity-js#2312] [How to... ?] [answered]
I am trying to use a DeleteRequest and am getting an error about my token being invalid, however when I debug into the Task, the token looks ok to me.   Is there a way I can debug the actual request that is being sent out so I can verify the headers/body?

```
export const deleteResource = (url: string): Task => {
  return Task.where("Delete the resource",
    Send.a(DeleteRequest.to(url)
      .using({
        headers: {
          'Accept': 'application/prs.hal-forms+json',
          'Authorization': "Bearer " + notes().get('access_token'),
          'Content-Type': 'application/json',
        }
      })),
    Debug.values((results, response, token) => {
        console.log(response);
      },
      LastResponse.body(),
      notes().get('access_token'),
    ),
    Ensure.that(LastResponse.status(), equals(204)),
  )
}
```

**Comment by lostdog19** [comment, marked as answer]
Answering my own question.   In my Actors class which implements Cast:

```
export class Actors implements Cast {
  constructor(private readonly page: Page) {
  }

  prepare(actor: Actor) {

    const axiosInstance = createAxios({
      timeout: 10 * 1000,
    });

    axiosInstance.interceptors.request.use(request => {
      console.log("Starting Request", request);
      return request;
    })

    axiosInstance.interceptors.response.use(response => {
      console.log("Response:", response);
      return response;
    })

    return actor.whoCan(CallAnApi.using(axiosInstance))
  }

  static initialize(page: Page) {
    engage(new Actors(page));
  }
}
```

**Reply by lostdog19** [reply]
Not sure if this is the best way, but it appears to work.

**Reply by jan-molak** [reply]
Sure, using interceptors makes sense. Also, when an actor performs an interaction to [`Send`](https://serenity-js.org/api/rest/class/Send/), they should [automatically collect the request/response pair](https://github.com/serenity-js/serenity-js/blob/2f09f80783e3189088335a8e44de013199209d9e/packages/rest/src/screenplay/interactions/Send.ts#L64) and attach it to the Serenity BDD report.

Is this not what you're seeing in your scenario?
